service/controller/v2/resource/chart/update.go

```diff
@@ -95,7 +94,7 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the chart has to be updated")
 
-	if currentChartState.ReleaseVersion != "" && !reflect.DeepEqual(desiredChartState, currentChartState) {
```

Comment:
Bug meant this was always returning true. The release status is set for the current state but not the desired state.

Reply:
Very good catch!